[rush] rush build error logs no longer include stdout

## Summary
`rush build` error logs no longer include STDOUT since `5.124.4`.

## Repro steps
Using https://github.com/microsoft/rush-example
1. Add TS error in package
2. `rush build`
3. Review generated logs

**Expected result:** (v5.124.4)
```
==[ FAILURE: 1 operation ]=====================================================

--[ FAILURE: my-toolchain ]----------------------------------[ 0.77 seconds ]--

Invoking: rimraf ./lib/ && tsc 
src/startCommand.ts(9,1): error TS2552: Cannot find name 'consoleX'. Did you mean 'console'?


Operations failed.

rush build (0.79 seconds)
```

**Actual result:** (since v5.124.4)
```
==[ FAILURE: 1 operation ]=====================================================

--[ FAILURE: my-toolchain ]----------------------------------[ 0.76 seconds ]--

Returned error code: 2


Operations failed.

rush build (0.77 seconds)
```

## Details
Potentially introduced by https://github.com/microsoft/rushstack/pull/4717

[Zulip Thread](https://rushstack.zulipchat.com/#narrow/stream/262513-general/topic/rush.20build.20print.20stdout.20.26.20stderr.20without.20trimming)

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | <!-- X.Y.Z --> |
| `rushVersion` from rush.json? | 5.124.4 vs 5.124.5 |
| `useWorkspaces` from rush.json? | true |
| Operating system? | Mac |
| Would you consider contributing a PR? | Yes |
| Node.js version (`node -v`)? | 20.11.0 |

--- a/common/changes/@microsoft/rush/stdio-blank-summary_2024-10-29-00-31.json
+++ b/common/changes/@microsoft/rush/stdio-blank-summary_2024-10-29-00-31.json
@@ -1,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Fix an issue where if an Operation wrote all logs to stdout, then exited with a non-zero exit code, only the non-zero exit code would show up in the summary.",
+      "type": "none"
+    }
+  ],
+  "packageName": "@microsoft/rush"
+}

--- a/libraries/rush-lib/src/logic/operations/OperationExecutionManager.ts
+++ b/libraries/rush-lib/src/logic/operations/OperationExecutionManager.ts
@@ -289,10 +289,12 @@ export class OperationExecutionManager {
     if (message) {
       // This creates the writer, so don't do this until needed
       record.collatedWriter.terminal.writeStderrLine(message);
-      // Ensure that the error message, if present, shows up in the summary
+      // Ensure that the summary isn't blank if we have an error message
+      // If the summary already contains max lines of stderr, this will get dropped, so we hope those lines
+      // are more useful than the final exit code.
       record.stdioSummarizer.writeChunk({
         text: `${message}\n`,
-        kind: TerminalChunkKind.Stderr
+        kind: TerminalChunkKind.Stdout
       });
     }
   }

